pkg/api/authz/tool.go

```diff
@@ -2,14 +2,17 @@ package authz
 
 import (
 	"net/http"
+	"strings"
 
 	"github.com/obot-platform/nah/pkg/router"
 	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
 	"k8s.io/apiserver/pkg/authentication/user"
 )
 
 func (a *Authorizer) checkTools(req *http.Request, resources *Resources, _ user.Info) (bool, error) {
-	if resources.ToolID == "" {
+	// Skip this auth check if the request isn't for a custom tool
+	if resources.ToolID == "" || !strings.HasPrefix(resources.ToolID, system.ToolPrefix) {
```

Comment:
If this isn't a custom tool, does it make sense to get the `ToolReference` and check to see if that tool is available for the agent or obot?

Reply:
I'm not sure. This was the solution Darren suggested here, but you may have a point. IIUC, the current implementation would allow you to call deauthenticate (as well as other endpoints) for tools that may not be available on the agent/obot. AFAICT, this was the original behavior before custom tools went in, so may have been a pre-existing gap.

Reply:
The endpoint you're fixing here also allows a comma-separated list of tools, too. So this is probably the right thing for now.